File: C++/A/_1008A-romaji.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    string vowels = "aeiou", word;
    cin >> word;
    bool flag = true;
    for (int i = 0; i < word.size() - 1; i++)
    {
        if (word[i] == 'n')
            continue;
        if (vowels.find(word[i]) == string::npos && vowels.find(word[i + 1]) == string::npos)
            flag == false;
    }
    cout << ((flag) ? "Yes\n" : "No\n");
}

File: Python/A/_1008A-romaji.py
def is_berlanese(word):
    vowels = "aeiou"
    for i in range(len(word)):
        if word[i] not in vowels and word[i] != 'n':
            if i + 1 < len(word) and word[i + 1] not in vowels:
                return "NO"
    return "YES"


word = input()
if len(word) == 1:
    print("YES" if (word in "aeinou") else "NO")
elif len(word) == 2:
    if word[1] not in "aeinou":
        print("NO")
    else:
        print("YES" if (word[0] in "aeinou") else "NO")
else:
    print(is_berlanese(word))
